astroquery/xmm_newton/core.py

```diff
@@ -0,0 +1,343 @@
+        """
+        Download data from XMM-Newton
+
+        Parameters
+        ----------
+        observation_id : string
+            id of the observation to be downloaded, mandatory
+            The identifier of the observation we want to retrieve, 10 digits
+            example: 0144090201
+        level : string
+            level to download, optional, by default everything is downloaded
+            values: ODF, PPS
+        instname : string
+            instrument name, optional, two characters, by default everything
+            values: OM, R1, R2, M1, M2, PN
+        instmode : string
+            instrument mode, optional
+            examples: Fast, FlatFieldLow, Image, PrimeFullWindow
+        filter : string
+            filter, optional
+            examples: Closed, Open, Thick, UVM2, UVW1, UVW2, V
+        expflag : string
+            exposure flag, optional, by default everything
+            values: S, U, X(not applicable)
+        expno : integer
+            exposure number with 3 digits, by default all exposures
+            examples: 001, 003
+        name : string
+            product type, optional, 6 characters, by default all product types
+            examples: 3COLIM, ATTTSR, EVENLI, SBSPEC, EXPMAP, SRCARF
+        datasubsetno : character
+            data subset number, optional, by default all
+            examples: 0, 1
+        sourceno : hex value
+            source number, optional, by default all sources
+            example: 00A, 021, 001
+        extension : string
+            file format, optional, by default all formats
+            values: ASC, ASZ, FTZ, HTM, IND, PDF, PNG
+        filename : string
+            file name to be used to store the file, optional, default
+            None
+        verbose : bool
+            optional, default 'False'
+            flag to display information about the process
+
+        Returns
+        -------
+        None. It downloads the observation indicated
+        """
+
+        link = self.data_aio_url + "obsno=" + observation_id
+
+        if level is not None:
+            link = link + "&level=" + level
+
+        if instname is not None:
+            link = link + "&instname=" + instname
+
+        if instmode is not None:
+            link = link + "&instmode=" + instmode
+
+        if filter is not None:
+            link = link + "&filter=" + filter
+
+        if expflag is not None:
+            link = link + "&expflag=" + expflag
+
+        if expno is not None:
+            link = link + "&expno=" + expno
+
+        if name is not None:
+            link = link + "&name=" + name
+
+        if datasubsetno is not None:
+            link = link + "&datasubsetno=" + datasubsetno
+
+        if sourceno is not None:
+            link = link + "&sourceno=" + sourceno
+
+        if extension is not None:
+            link = link + "&extension=" + extension
+
+        response = self._handler.request('GET', link)
```

Comment:
Why not just `response = self._request('GET', link)` ?  

But more importantly, why not follow the example of other astroquery modules, e.g.:
```
        response = self._request('GET', url, params=payload,
                                 timeout=self.TIMEOUT,
                                 cache=cache)
```

Especially note the `params=payload` section.  If all of the arguments to be passed to the server are key-value pairs anyway, this is the _correct_ way to URL-format them.  You should not be manually formatting URLs with `&` separators.

Reply:
Ok. Now directly to response = self._request('GET', link)

Reply:
It has been changed to:
response = self._request('GET', link)